modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/transaction/DefaultTransactionHandle.java

```diff
@@ -6,22 +6,47 @@
  */
 package org.mule.runtime.module.extension.internal.runtime.transaction;
 
-import org.mule.runtime.extension.api.connectivity.TransactionalConnection;
+import org.mule.runtime.api.tx.TransactionException;
+import org.mule.runtime.core.api.transaction.Transaction;
+import org.mule.runtime.core.api.transaction.TransactionCoordination;
 import org.mule.runtime.extension.api.tx.TransactionHandle;
-import org.mule.runtime.extension.api.tx.Transactional;
 
 /**
- * Default implementation of {@link TransactionHandle}. Use this instance when an actual transaction <b>has</b> been
- * started
+ * Default {@link TransactionHandle} implementation.
+ * <p>
+ * This will execute a transaction action only is the transaction is still alive.
+ *
+ * @since 4.0
  */
-public class DefaultTransactionHandle extends IdempotentTransactionHandle<TransactionalConnection> {
+public class DefaultTransactionHandle implements TransactionHandle {
 
   /**
-   * Creates a new instance
-   *
-   * @param connection the connection on which the transaction started
+   * {@inheritDoc}
    */
-  public DefaultTransactionHandle(TransactionalConnection connection) {
-    super(connection, Transactional::commit, Transactional::rollback);
+  @Override
+  public boolean isTransacted() {
+    return true;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void commit() throws TransactionException {
+    Transaction transaction = TransactionCoordination.getInstance().getTransaction();
+    if (transaction != null) {
```

Comment:
optional if present

Reply:
is unnecessary and also can't be done because `commit()` and `rollback()` throws checked exceptions